dts/arm/nordic/nrf52840.dtsi

```diff
@@ -55,6 +55,31 @@
 			label = "UART_1";
 		};
 
+		gpiote0: gpiote@40006000 {
+			compatible = "nordic,nrf5-gpiote";
+			reg = <0x40006000 0x1000>;
+			status = "disabled";
+			label = "GPIOTE_0";
+		};
```

Comment:
If gpiote channel configuration is done in a gpiote driver along with reference to a specific gpio pin,
I'd expose that in the following way, as a subnode of gpiote
```
                gpio0pin1event0: channel@510 {
                          reg = <0x510>;
                          gpio = <&gpio0 1>;
                          mode = "event";
                          polarity = "HiToLo";
                }
```
Though, if we consider each gpiote if associated to a gpio (since IIUC we cannot configure GPIO port in GPIOTE register), gpiote0 can maybe be set as a subnode of gpio0:
```
		gpio0: gpio@50000000 {
			compatible = "nordic,nrf5-gpio";
			gpio-controller;
			reg = <0x50000000 0x1000>;
			#gpio-cells = <2>;
			label = "GPIO_0";
			status = "disabled";

		         gpiote0: gpiote@40006000 {
			         compatible = "nordic,nrf5-gpiote";
			         reg = <0x40006000 0x1000>;
			         status = "disabled";
			         label = "GPIOTE_0";

                                gpio0pin1event0: channel@510 {
                                            reg = <0x510>;
                                            mode = "event";
                                            polarity = "HiToLo";
                               };

         		};
		};
```

Reply:
@pizi-nordic, @anangl, @kl-cruz  mind to comment on the above?

Reply:
@erwango From what I can tell GPIO and GPIOTE are completely separate, independent peripherals. When there's a single port (nRF52832 for example) then one selects the pin with `PSEL` in GPIOTE in order to configure an interrupt or task/event connection. If there's more than one port (nRF52840), one has to configure both the port and the pin in the GPIOTE config.
http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52832.ps.v1.1/gpiote.html?cp=2_1_0_20_3_2#register.CONFIG-0
http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52840.ps/gpiote.html?cp=2_0_0_5_9_3_2#register.CONFIG-0-7

Reply:
Ok, I had seen only the on port specification.
Then I think following would make sense as an individual node (not gpio subnode)
```
		         gpiote0: gpiote@40006000 {
			         compatible = "nordic,nrf5-gpiote";
			         reg = <0x40006000 0x1000>;
			         status = "disabled";
			         label = "GPIOTE_0";

                                gpio0pin1event0: channel@510 {
                                            reg = <0x510>;
                                            gpios = <&gpio0 1>
                                            mode = "event";
                                            polarity = "HiToLo";
                               };
```

One thing to check is the gpio-cells dimension (here I assumed gpio-cells = 1), as generally this is 2 but last slot is used for IRQ (which is not available here if IUC). But it could be replaced with any other information pertinent to gpio way of working.

Reply:
At the moment the gpio-cells is 2, for pin and flags.

```
                            gpiote0: gpiote@40006000 {
			        compatible = "nordic,nrf5-gpiote";
			        reg = <0x40006000 0x1000>;
			        status = "disabled";
			        label = "GPIOTE_0";

                                gpiote_chan0: channel@510 {
                                            reg = <0x510>;
                                            status = "disabled";
                                };

                                gpiote_chan1: channel@514 {
                                            reg = <0x514>;
                                            status = "disabled";
                                };
```

.. and then in the board:
```
    &gpiote_chan0 {
        gpios = <&gpio0 1 0>
        mode = "event";
        polarity = "HiToLo";
        status = "ok";
    };
```

That said, I'm not sure how this will all work to enable other drivers (for example an I2C sensor) to specify a gpio for interrupt source.

Reply:
> That said, I'm not sure how this will all work to enable other drivers (for example an I2C sensor) to specify a gpio for interrupt source.

You should add, in gptiote:
```
			interrupt-controller;
			#interrupt-cells = <x>;
```
In the node that need an irq:
```
			interrupt-parent = <&gpiote_chan0>;
			interrupts = <1>;
			interrupt-names = "bla";
```
Note that I am unsure how the interrupts should be populated when you have only one line.
Maybe you can remove the channels subnode and do, in i2c slave node, something like:
```
			interrupt-parent = <&gpiote>;
			interrupts = <1 2 3>;
			interrupt-names = "bla";
```
with, intterupt-cells = <3> in gpiote that could hold the following fields
1: channel
2: event
3: polarity